dice_ml/diverse_counterfactuals.py

```diff
@@ -135,10 +135,16 @@ def display_df(self, df, show_only_changes):
             org = self.test_instance_df.values.tolist()[0]
             for ix in range(df.shape[0]):
                 for jx in range(len(org)):
-                    if math.isclose(newdf[ix][jx] == org[jx], rel_tol=org[jx]/10000):
-                        newdf[ix][jx] = '-'
+                    if not isinstance(newdf[ix][jx], str):
+                        if math.isclose(newdf[ix][jx], org[jx], rel_tol=abs(org[jx]/10000)):
+                            newdf[ix][jx] = '-'
+                        else:
+                            newdf[ix][jx] = str(newdf[ix][jx])
                     else:
-                        newdf[ix][jx] = str(newdf[ix][jx])
+                        if newdf[ix][jx] == org[jx]:
```

Comment:
Makes sense. This was the original code. We retain the original code for str values but change it only for numeric values.